### PR TITLE
Test: permissioned strategy

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
Adds tests for the permissioned strategy being used for the initial vaults.

# Changes
- Permissioned strategy tests
- Fixes bug in `returnAssets` function
- Fixes whitelisting bug in constructor

### CODE REVIEW

File: contracts/strategies/PermissionedStrategy.sol [hunk 1]
```diff
@@ -35,11 +35,12 @@ contract PermissionedStrategy is IStrategy, AccessControl {
...
     ) {
         // whitelist default assets
-        setAssetWhitelist(poolShortToken, true);
-        setAssetWhitelist(vaultAsset, true);
+        assetWhitelist[poolShortToken] = true;
+        assetWhitelist[vaultAsset] = true;
 
         // make contract deployer the default admin
         _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(WHITELISTER, msg.sender);
```
Comment (0xKoda) [inline contracts/strategies/PermissionedStrategy.sol:43]: Should we have a function to change this auth later down the line?

**Review by 0xKoda — APPROVED** [review]
Looks good

File: contracts/strategies/PermissionedStrategy.sol (hunk 1 above)
Comment (0xKoda) [reply contracts/strategies/PermissionedStrategy.sol:43]: (incase we wish to migrate)

Comment (raymogg) [reply contracts/strategies/PermissionedStrategy.sol:43]: The default admin is able to change other roles. Let me verify that the OZ package comes with these functions by default but I am fairly sure it does.

**Pull request merged**